cmake build fails from release tgz

I tried a cmake build on MacOS. While it worked when I used it directly on checked out git sources, trying to build syslog-ng using a tgz built from the same git snapshot, the build fails in the configuration phase. One problem is the freshly introduced gperf (https://github.com/syslog-ng/syslog-ng/pull/4763), the other one is Python, which is especially interesting, as all Python support is turned off:

```
bash-3.2$ cmake --install-prefix    $PWD/build/install      -B build .  -DCMAKE_BUILD_TYPE=Debug  -DBUILD_TESTING=OFF -DIVYKIS_SOURCE=system -DPYTHON_VERSION=3 -DENABLE_AFSMTP=OFF  -DENABLE_SNMP=OFF  -DENABLE_JAVA=OFF -DENABLE_JAVA_MODULES=OFF -DENABLE_PYTHON=OFF -DENABLE_PYTHON_MODULES=OFF -DENABLE_MQTT=OFF -DENABLE_PACCT=OFF -DENABLE_CPP=OFF -DENABLE_EXAMPLE_MODULES=OFF   -DSUMMARY_VERBOSE=ON --fresh       -DENABLE_BIGQUERY=OFF
-- The C compiler identification is AppleClang 15.0.0.15000100
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:43 (include):
  include could not find requested file:

    FindGperf


-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of struct sockaddr_storage
-- Check size of struct sockaddr_storage - done
-- Check size of struct ucred
-- Check size of struct ucred - failed
-- Check size of struct cmsgcred
-- Check size of struct cmsgcred - failed
-- Performing Test SYSLOG_NG_HAVE_CTRLBUF_IN_MSGHDR
-- Performing Test SYSLOG_NG_HAVE_CTRLBUF_IN_MSGHDR - Success
-- Check size of struct sockaddr_in6
-- Check size of struct sockaddr_in6 - done
-- Looking for TCP_KEEPIDLE
-- Looking for TCP_KEEPIDLE - not found
-- Looking for TCP_KEEPINTVL
-- Looking for TCP_KEEPINTVL - found
-- Looking for TCP_KEEPCNT
-- Looking for TCP_KEEPCNT - found
-- Check size of O_LARGEFILE
-- Check size of O_LARGEFILE - failed
-- Performing Test SYSLOG_NG_HAVE_THREAD_KEYWORD
-- Performing Test SYSLOG_NG_HAVE_THREAD_KEYWORD - Success
-- Looking for strtoll
-- Looking for strtoll - found
-- Looking for strnlen
-- Looking for strnlen - found
-- Looking for getline
-- Looking for getline - found
-- Looking for strtok_r
-- Looking for strtok_r - found
-- Looking for strtoimax
-- Looking for strtoimax - found
-- Looking for inet_aton
-- Looking for inet_aton - found
-- Looking for inet_ntoa
-- Looking for inet_ntoa - found
-- Looking for getutent
-- Looking for getutent - not found
-- Looking for getutxent
-- Looking for getutxent - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for getprotobynumber_r
-- Looking for getprotobynumber_r - not found
-- Looking for clock_gettime
-- Looking for clock_gettime - found
-- Looking for gmtime_r
-- Looking for gmtime_r - found
-- Looking for localtime_r
-- Looking for localtime_r - found
-- Looking for getrandom
-- Looking for getrandom - not found
-- Looking for fmemopen
-- Looking for fmemopen - found
-- Looking for memrchr
-- Looking for memrchr - not found
-- Looking for strcasestr
-- Looking for strcasestr - found
-- Looking for pread
-- Looking for pread - found
-- Looking for pwrite
-- Looking for pwrite - found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - not found
-- Looking for timezone
-- Looking for timezone - found
-- Looking for include file utmp.h
-- Looking for include file utmp.h - found
-- Looking for include file utmpx.h
-- Looking for include file utmpx.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file getopt.h
-- Looking for include file getopt.h - found
-- Performing Test UTMPX_HAS_UT_TYPE
-- Performing Test UTMPX_HAS_UT_TYPE - Success
-- Performing Test UTMP_HAS_UT_TYPE
-- Performing Test UTMP_HAS_UT_TYPE - Failed
-- Performing Test UTMPX_HAS_UT_USER
-- Performing Test UTMPX_HAS_UT_USER - Success
-- Performing Test UTMP_HAS_UT_USER
-- Performing Test UTMP_HAS_UT_USER - Failed
-- Performing Test SYSLOG_NG_HAVE_STRUCT_TM_TM_GMTOFF
-- Performing Test SYSLOG_NG_HAVE_STRUCT_TM_TM_GMTOFF - Success
-- Looking for SO_MEMINFO
-- Looking for SO_MEMINFO - not found
-- Looking for linux/sock_diag.h
-- Looking for linux/sock_diag.h - not found
-- Found BISON: /usr/local/opt/bison/bin/bison (found suitable version "3.8.2", minimum required is "3.7.6") 
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
-- Found GLIB: /usr/local/Cellar/glib/2.78.3/include/glib-2.0;/usr/local/Cellar/glib/2.78.3/lib/glib-2.0/include (found suitable version "2.78.3", minimum required is "2.32.0") 
-- Looking for g_list_copy_deep
-- Looking for g_list_copy_deep - found
-- Looking for g_ptr_array_find_with_equal_func
-- Looking for g_ptr_array_find_with_equal_func - found
-- Looking for g_canonicalize_filename
-- Looking for g_canonicalize_filename - found
CMake Warning at cmake/Modules/LibFindMacros.cmake:299 (message):
  WARNING: MISSING PACKAGE

  We could not find development headers for RabbitMQ.  Do you have the
  necessary dev package installed? This package is NOT REQUIRED and you may
  ignore this warning but by doing so you may miss some functionality of
  syslog-ng.

  Relevant CMake configuration variables:

    RabbitMQ_INCLUDE_DIR=<not found>
    RabbitMQ_LIBRARY=<not found>

  You may use CMake GUI, cmake -D or ccmake to modify the values.  Delete
  CMakeCache.txt to discard all values and force full re-detection if
  necessary.

Call Stack (most recent call first):
  cmake/Modules/FindRabbitMQ.cmake:29 (libfind_process)
  CMakeLists.txt:243 (find_package)


-- Found IVYKIS 
-- Looking for iv_work_pool_submit_continuation
-- Looking for iv_work_pool_submit_continuation - not found
-- Found OpenSSL: /usr/local/Cellar/openssl@3/3.2.0_1/lib/libcrypto.dylib (found version "3.2.0")  
-- Found FLEX: /usr/bin/flex (found version "2.6.4") 
CMake Warning (dev) at cmake/Modules/FindLIBNET.cmake:44 (EXEC_PROGRAM):
  Policy CMP0153 is not set: The exec_program command should not be called.
  Run "cmake --help-policy CMP0153" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Use execute_process() instead.
Call Stack (most recent call first):
  CMakeLists.txt:259 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at cmake/Modules/FindLIBNET.cmake:45 (EXEC_PROGRAM):
  Policy CMP0153 is not set: The exec_program command should not be called.
  Run "cmake --help-policy CMP0153" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Use execute_process() instead.
Call Stack (most recent call first):
  CMakeLists.txt:259 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found LIBNET: -L/usr/local/Cellar/libnet/1.3/lib -lnet  
-- Could NOT find WRAP (missing: WRAP_LIBRARY WRAP_INCLUDE_DIR) 
-- Found CRITERION 
-- Checking for one of the modules 'libsystemd>=245'
-- Found lib version 
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
CMake Warning (dev) at CMakeLists.txt:306 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonInterp: /usr/local/bin/python3 (found suitable exact version "3.11.7") 
CMake Warning (dev) at CMakeLists.txt:307 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PythonLibs: /usr/local/Frameworks/Python.framework/Versions/3.11/lib/libpython3.11.dylib (found suitable exact version "3.11.7") 
-- Detected pythonlib version: 3.11.7
CMake Error at CMakeLists.txt:311 (include):
  include could not find requested file:

    python_build_venv


-- Looking for EVP_MD_CTX_reset
-- Looking for EVP_MD_CTX_reset - found
-- Looking for ASN1_STRING_get0_data
-- Looking for ASN1_STRING_get0_data - found
-- Looking for SSL_CTX_get0_param
-- Looking for SSL_CTX_get0_param - found
-- Looking for X509_STORE_CTX_get0_cert
-- Looking for X509_STORE_CTX_get0_cert - found
-- Looking for X509_get_extension_flags
-- Looking for X509_get_extension_flags - found
-- Looking for DH_set0_pqg
-- Looking for DH_set0_pqg - found
-- Looking for BN_get_rfc3526_prime_2048
-- Looking for BN_get_rfc3526_prime_2048 - found
-- Looking for SSL_CONF_CTX_new
-- Looking for SSL_CONF_CTX_new - found
-- Looking for SSL_CTX_set_num_tickets
-- Looking for SSL_CTX_set_num_tickets - found
-- Looking for SSL_CTX_set_ciphersuites
-- Looking for SSL_CTX_set_ciphersuites - found
-- Looking for SSL_CTX_set1_sigalgs_list
-- Looking for SSL_CTX_set1_sigalgs_list - found
-- Looking for SSL_CTX_set1_client_sigalgs_list
-- Looking for SSL_CTX_set1_client_sigalgs_list - found
-- Looking for SSL_CTX_set_min_proto_version
-- Looking for SSL_CTX_set_min_proto_version - found
-- Looking for SSL_add_file_cert_subjects_to_stack
-- Looking for SSL_add_file_cert_subjects_to_stack - found
-- Looking for SSL_add_dir_cert_subjects_to_stack
-- Looking for SSL_add_dir_cert_subjects_to_stack - found
-- Checking for module 'libpcre2-8'
--   Found libpcre2-8, version 10.42
CMake Error at lib/CMakeLists.txt:384 (gperf_generate):
  Unknown CMake command "gperf_generate".


-- Configuring incomplete, errors occurred!

```

I just ran into this same issues while attempting to bring `syslog-ng` up to date for MacPorts.

I managed to work around the `FindGperf` issue by installing the `kde-extra-cmake-modules` port, and `Unknown CMake command "gperf_generate"` by patching `lib/CMakeLists.txt` to change `gperf_generate` to `ecm_gperf_generate`

However, to deal with
```
include could not find requested file:

    python_build_venv
```
I tried commenting out `include(python_build_venv)`. This allowed cmake to finish cleanly.

While building I ran into a new issue:

```
:info:build [ 81%] Building C object modules/afsnmp/CMakeFiles/afsnmp.dir/afsnmpdest.c.o
:info:build [ 81%] Built target stardate
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f modules/tagsparser/CMakeFiles/tags-parser.dir/build.make modules/tagsparser/CMakeFiles/tags-parser.dir/depend
:info:build cd /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/modules/afsnmp && /usr/bin/clang -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_68 -D_GNU_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -D__APPLE_USE_RFC_3542 -Dafsnmp_EXPORTS -I/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build -I/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/modules/afsnmp -I/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/modules/afsnmp -I/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/lib -I/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/lib -isystem /opt/local/include -isystem /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/ivykis-install/include -isystem /opt/local/include/json-c -isystem /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/lib/eventlog/src -isystem /opt/local/include/glib-2.0 -isystem /opt/local/lib/glib-2.0/include -isystem /opt/local/libexec/openssl3/include -pipe -Os -DNDEBUG -I/opt/local/include -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -std=gnu99 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -mmacosx-version-min=14.0 -fPIC -DNETSNMP_ENABLE_IPV6 -fno-strict-aliasing -g -Os -pipe -DBUILD=177 -DMACOSX_DEPLOYMENT_TARGET=14.5 --sysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -iframeworkwithsysroot /System/Library/Frameworks/-f985-11ee-b563-fe8bc7981bff/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.Internal.sdk/System/Library/PrivateFrameworks/ -f985-11ee-b563-fe8bc7981bff/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.Internal.sdk/System/Library/Frameworks/ -iwithsysroot /usr/local/libressl/include -L/AppleInternal/Library/BuildRoots/91a344b1-f985-11ee-b563-fe8bc7981bff/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.Internal.sdk/usr/local/libressl/lib -Udarwin23 -Ddarwin23=darwin23 -I. -I/usr/local/include -Wshadow -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -Wimplicit-function-declaration -Wnested-externs -Wstrict-prototypes -Wswitch-default -Wimplicit-int -Wall -Wuninitialized -Wdeprecated -Wdeprecated-declarations -Woverflow -Wdouble-promotion -Wfloat-equal -Wpointer-arith -Wpointer-sign -Wmissing-format-attribute -Wold-style-definition -Wundef -Wignored-qualifiers -Wfloat-conversion -Wbad-function-cast -MD -MT modules/afsnmp/CMakeFiles/afsnmp.dir/afsnmpdest.c.o -MF CMakeFiles/afsnmp.dir/afsnmpdest.c.o.d -o CMakeFiles/afsnmp.dir/afsnmpdest.c.o -c /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/modules/afsnmp/afsnmpdest.c
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build'
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build'
:info:build /Library/Developer/CommandLineTools/usr/bin/make  -f modules/system-source/CMakeFiles/system-source.dir/build.make modules/system-source/CMakeFiles/system-source.dir/build
:info:build make[2]: Entering directory `/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build'
:info:build [ 81%] Generating tags-parser-grammar.y
:info:build cd /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/modules/tagsparser && /opt/local/bin/python3 /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/lib/merge-grammar.py /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/modules/tagsparser/tags-parser-grammar.ym > /opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build/modules/tagsparser/tags-parser-grammar.y
:info:build clang: error: unknown argument: '-f985-11ee-b563-fe8bc7981bff/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.Internal.sdk/System/Library/Frameworks/'
:info:build make[2]: *** [modules/afsnmp/CMakeFiles/afsnmp.dir/afsnmpdest.c.o] Error 1
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/build'
:info:build make[1]: *** [modules/afsnmp/CMakeFiles/afsnmp.dir/all] Error 2

```

Note: I also set `-DENABLE_AFSMTP=OFF`, which was ignored, just like `-DENABLE_PYTHON=OFF`

The above error can be worked around by setting `-DENABLE_AFSNMP=OFF`. However, `Criterion` seems not to be an optional package as the build now fails with 

```
/opt/local/var/macports/build/_Users_marius_Development_MacPorts_ports_sysutils_syslog-ng/syslog-ng/work/syslog-ng-4.7.1/libtest/msg_parse_lib.c:28:10: fatal error: 'criterion/criterion.h' file not found
#include <criterion/criterion.h>
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [libtest/CMakeFiles/libtest.dir/msg_parse_lib.c.o] Error 1
```
 even though `cmake` couldn't find `Criterion` and I had set `-DBUILD_TESTING=OFF`

--- a/Makefile.am
+++ b/Makefile.am
@@ -162,6 +162,11 @@ local-check:
 				TESTS="${current_tests} ${subdir_tests}"
 endif
 
+# We use VERSION.txt, as VERSION file can cause C++ build errors on case-insensitive file-systems.
+# Some step(s?) still can produce or leave behind a VERSION file, clean it up before packing the final dist tar.gz file.
+dist-hook:
+	rm -f VERSION
+
 check: check_target_guard
 
 check_target_guard:

--- a/cmake/Makefile.am
+++ b/cmake/Makefile.am
@@ -13,6 +13,7 @@ EXTRA_DIST +=	\
 	cmake/Modules/FindFLEX.cmake	\
 	cmake/Modules/FindGettext.cmake	\
 	cmake/Modules/FindGLIB.cmake	\
+	cmake/Modules/FindGperf.cmake	\
 	cmake/Modules/FindGradle.cmake	\
 	cmake/Modules/FindgRPC.cmake	\
 	cmake/Modules/FindHiredis.cmake	\
@@ -36,4 +37,5 @@ EXTRA_DIST +=	\
 	cmake/Modules/ProtobufGenerateCpp.cmake	\
 	cmake/module_switch.cmake		\
 	cmake/openssl_functions.cmake \
-	cmake/print_config_summary.cmake
+	cmake/print_config_summary.cmake \
+	cmake/python_build_venv.cmake

--- a/modules/afmongodb/Makefile.am
+++ b/modules/afmongodb/Makefile.am
@@ -41,7 +41,8 @@ BUILT_SOURCES					+=	\
 
 EXTRA_DIST					+=	\
 	modules/afmongodb/afmongodb-grammar.ym \
-	modules/afmongodb/CMakeLists.txt
+	modules/afmongodb/CMakeLists.txt \
+	modules/afmongodb/tests/CMakeLists.txt
 
 
 .PHONY: modules/afmongodb/ mod-afmongodb mod-mongodb

